sql/wsrep_sst.cc

```diff
@@ -897,14 +1058,42 @@ static void *sst_donor_thread(void *a) {
                         // operate with wsrep_ready == OFF
   thd.ptr->wsrep_sst_donor = true;
 
+  // Create the SST auth user
+  err = wsrep_create_sst_user(true, password.c_str());
+
   // Launch the SST script and save pointer to its process:
 
   if (mysql_mutex_lock(&LOCK_wsrep_sst)) abort();
-  wsp::process proc(arg->cmd, "r", arg->env);
+  wsp::process proc(arg->cmd, "rw", arg->env, !err /* execute_immediately */);
+
+  if (!err)
+  {
+    // Pass the sst username/password as name/value pairs
+    int ret;
+    ret = snprintf(out_buf, out_len,
+                  "sst_user=mysql.pxc.sst.user\n"
+                  "sst_password=%s\n",
+                  password.c_str());
+    if (ret < 0 || ret >= out_len)
+    {
+      WSREP_ERROR("sst_donor_thread(): snprintf() failed: %d", ret);
+      err = (ret < 0 ? ret : -EMSGSIZE);
+    }
+
+    if (!err)
```

Comment:
is pipe communication secure ?

Reply:
https://unix.stackexchange.com/questions/156859/is-the-data-transiting-through-a-pipe-confidential